<launch>
  <arg name="sec_idle" default="3.0" />
  <arg name="use_base_odom" default="true" />

  <node pkg="marvin_controller" name="marvin_base_controller"
        type="base_controller.py" output="screen" ns="/marvin">
    <param name="sec_idle" value="$(arg sec_idle)" />
    <param name="publish_odom" value="$(arg use_base_odom)" />
    <remap from="marvin/cmd_vel" to="/marvin/cmd_vel" />
    <remap from="odom" to="/odom" />
  </node>
  <group if="$(arg use_base_odom)" >
    <node pkg="robot_pose_ekf" type="robot_pose_ekf" name="robot_pose_ekf">
      <param name="output_frame" value="odom"/>
      <param name="freq" value="30.0"/>
      <param name="sensor_timeout" value="1.0"/>
      <param name="odom_used" value="true"/>
      <param name="imu_used" value="true"/>
      <param name="vo_used" value="false"/>
      <param name="debug" value="true"/>
      <param name="self_diagnose" value="true"/>
    </node>
  </group>
  <group>
   <node pkg="laser_scan_matcher" type="laser_scan_matcher_node"
         name="laser_scan_matcher_node" output="screen">
     <param name="fixed_frame" value="odom"/>
     <param name="base_frame" value="base_footprint"/>
     <remap from="scan" to="/marvin/laser/scan_filtered" />
     <param name="max_iterations" value="30"/>
     <param name="max_correspondence_dist" value="0.5" />
     <param name="max_angular_correction_deg" value="60.0" />
     <param name="outliers_maxPerc" value="0.8" />
     <param name="use_odom" value="false"/>
     <param name="use_imu" value="false"/>
     <param name="use_vel" value="false"/>
   </node>
  </group>
</launch>
